### ISSUE TITLE
export * as ns syntax with @babel/preset-typescript

### ISSUE THREAD

**Issue by Pytal** [babel/babel#11826]
## Bug Report
**Current behavior**
TypeScript 3.8 shipped with support for [`export * as ns` Syntax](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8/#export-star-as-namespace-syntax) which should correctly transpile the following:

**Input Code**
``` ts
export * as api from './api'
```

**Expected behavior**
Using the code above with just [`@babel/preset-typescript`](https://github.com/babel/babel/tree/main/packages/babel-preset-typescript) will throw an error.

In the meanwhile I have been using the [`@babel/plugin-proposal-export-namespace-from`](https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-export-namespace-from) plugin.

Ideally the functionality of the [`@babel/plugin-proposal-export-namespace-from`](https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-export-namespace-from) plugin could be merged into [`@babel/preset-typescript`](https://github.com/babel/babel/tree/main/packages/babel-preset-typescript).

**Comment by nicolo-ribaudo** [comment]
`export * as ns` isn't TS-specific syntax; it's a normal JavaScript proposal. We'll add it to `@babel/preset-env` once it becomes stable.

**Comment by Pytal** [comment]
Right, I though since this syntax was released to stable in TypeScript 3.8 it'd be nice to have it supported in `@babel/preset-typescript` as well

**Comment by JLHwung** [comment]
I can reproduce the error on [REPL](https://babel.dev/repl#?browsers=node%2012&build=&builtIns=false&spec=false&loose=true&code_lz=KYDwDg9gTgLgBAKjgQwM5wGYQpqEC2cARAHQD0WEJ-AVqkUA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=true&targets=&version=7.10.4&externalPlugins=)

> We'll add it to @babel/preset-env once it becomes stable, or to its shippedProposal option when browsers start supporting it by default.

`export * as ns` is added to ES2020, we should add `proposal-export-namespace-from to `preset-env`.

> it'd be nice to have it supported in @babel/preset-typescript as well

`preset-typescript` is meant for supporting TypeScript-only features, see https://github.com/babel/babel/issues/10690#issuecomment-552125194. That said, `preset-env` is still required to transform latest ECMAScript features.

**Comment by clar-cmp** [comment]
Also ran into this -- IMHO, `preset-typescript` should include proposals that aren't proper ECMAScript features yet. Right now, because it's included as part of ES2020, it makes sense to include in preset-env, but in the future I feel like clarifying something along these lines in the documentation for preset-typescript would be helpful.

**Comment by JLHwung** [comment]
> clarifying something along these lines in the documentation for preset-typescript would be helpful.

@clar-cmp I agree, can you send a PR on https://github.com/babel/website/blob/master/docs/preset-typescript.md ?

**Comment by clar-cmp** [comment]
I mean, what I'm proposing is technically a policy change, so I don't know if someone who isn't managing the project should just change the docs to say that.

**Comment by JLHwung** [comment]
Since `export-ns-from` has been included in `preset-env`, transferred this issue to `babel/website`. We should clarify that `@babel/preset-typescript` does not include stage 4 proposals.
